packages/router/src/util.ts

```diff
@@ -226,3 +226,39 @@ export {
   validatePath,
   replaceParams,
 }
+
+/**
+ * gets the announcement for the new page.
+ * called in page-loader's `componentDidUpdate`.
+ *
+ * the order of priority is:
+ * 1. RouteAnnouncement (the most specific one)
+ * 2. h1
+ * 3. document.title
+ * 4. location.pathname
+ */
+export const getAnnouncement = () => {
+  const routeAnnouncements = global?.document.querySelectorAll(
+    '[data-redwood-route-announcement]'
+  )
+
+  let routeAnnouncement
+
+  if (routeAnnouncements.length) {
+    routeAnnouncement = routeAnnouncements[
+      routeAnnouncements.length - 1
+    ] as HTMLDivElement
+  }
+
+  const pageHeading = global?.document.querySelector(`h1`)
+
+  if (routeAnnouncement && routeAnnouncement.textContent) {
+    return routeAnnouncement.textContent
+  } else if (pageHeading && pageHeading.textContent) {
+    return pageHeading.textContent
+  } else if (global?.document.title) {
+    return document.title
+  } else {
+    return `new page at ${global?.location.pathname}`
+  }
+}
```

Comment:
I suggestion to make this feel a bit more robust is to "return early", so find and validate the element your looking for, and return if it matches:


```js
export const getAnnouncement = () => {
   // find route annoucements element
  const ra = globals?.document.querySelectorAll( '[data-redwood-route-announcement]')?.[0]
  
  if (ra?.textContent) {
    return ra.textContent
  }
  
  /* ... */
```